compiler/src/iree/compiler/Codegen/Common/GPU/Passes.td

```diff
@@ -111,6 +111,22 @@ def GPUPromoteMatmulOperandsPass :
   ];
 }
 
+def GPUPropagateDispatchSizeBoundsPass :
+    InterfacePass<"iree-codegen-gpu-propagate-dispatch-size-bounds", "mlir::FunctionOpInterface"> {
+  let summary = "Pass to annotate workitem and workgroup IDs with known bounds";
+  let options = [
+    Option<"maxWorkgroupsX", "max-workgroups-x", "unsigned",
+           /*default=*/"0x7fffffff",
+           "Maximum number of workgroups in the x dimension permitted by the hardware">,
+    Option<"maxWorkgroupsYZ", "max-workgroups-yz", "unsigned",
+          /*default=*/"0x7fffffff",
+          "Maximum number of workgroups in the y and z dimension permitted by the hardware">,
+    Option<"maxWorkgroupSize", "max-workgroup-size", "unsigned",
+          /*default=*/"1024",
+          "Maximum workgroup size when none is specified">
```

Comment:
BTW, we already have some target device info like this in `hal.executable.variant` target info:
```
    hal.executable.variant public @rocm_hsaco_fb target(<"rocm", "rocm-hsaco-fb", {iree.gpu.target = #iree_gpu.target<arch = "gfx942", features = "", wgp = <compute =  fp64|fp32|fp16|int64|int32|int16|int8, storage =  b64|b32|b16|b8, subgroup =  shuffle|arithmetic, dot =  dp4xi8toi32, mma = [<MFMA_F16_16x16x16_F32>, <MFMA_F16_32x32x8_F32>], subgroup_size_choices = [64], max_workgroup_sizes = [1024, 1024, 1024], max_thread_count_per_workgroup = 1024, max_workgroup_memory_bytes = 65536>>, ukernels = "none", waves_per_eu = 2 : i64}>) {
```

Reply:
Good point, I could just get it from there

Reply:
+1 all of these values should be queried from the target information annotating the parent function rather than being pass options.